webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/DataSourceController.scala

```diff
@@ -439,6 +441,22 @@ class DataSourceController @Inject()(
       }
     }
 
+  def compose(token: Option[String]): Action[ComposeRequest] =
+    Action.async(validateJson[ComposeRequest]) { implicit request =>
+      val userToken = urlOrHeaderToken(token, request)
+      accessTokenService.validateAccess(UserAccessRequest.administrateDataSources(request.body.organizationName), token) {
+        for {
+          _ <- Fox.serialCombined(request.body.layers.map(_.datasetId).toList)(
+            id =>
+              accessTokenService.assertUserAccess(
+                UserAccessRequest.readDataSources(DataSourceId(id.name, id.owningOrganization)),
+                userToken))
+          _ <- composeService.composeDataset(request.body, userToken)
+          _ <- dataSourceService.checkInbox(verbose = false)
```

Comment:
this should also use `dataSourceRepository.updateDataSource(dataSource)` instead of `checkInbox`. (either return the dataSource out of the composeService or call that in there)